ecs-agent/go.mod

```diff
@@ -3,18 +3,35 @@ module github.com/aws/amazon-ecs-agent/ecs-agent
 go 1.19
 
 require (
+	github.com/aws/amazon-ecs-agent/agent v0.0.0-20230426070318-0202ad238404
```

Comment:
`agent` module depends on `ecs-agent` module - having `ecs-agent` depend on `agent` doesn't seem ideal. Can we avoid this dependency cycle?